x/mongo/driver/topology/connection.go

```diff
@@ -374,6 +374,17 @@ func (c *connection) readWireMessage(ctx context.Context, dst []byte) ([]byte, e
 		}
 	}
 
+	// In the case of an isMaster response where MaxMessageSize has not yet been set, use a hard-coded
+	// 48000 bytes instead.
+	maxMessageSize := c.desc.MaxMessageSize
+	if maxMessageSize == 0 {
```

Comment:
Wasn't quite sure how to check if we're reading the response of the initial "isMaster" here. In that case, `c.desc.MaxMessageSize` would have its zero-value, which would just be `0`. That's what I check for here, thinking that a user manually setting a `MaxMessageSize` of `0` on a server description would make no sense. Let me know if that's not true, or there's a better way of doing this check.

Reply:
This looks fine to me. An actual server shouldn't ever return 0 as it's max message size and defaulting to 48MB in that case seems reasonable.

Reply:
👍 , discussed offline: changed default to `48000000` instead of `48000` which was incorrect.